### PR TITLE
Updated CHANGELOGs and READMEs ahead of the Key Vault beta release for November 2021.

### PR DESCRIPTION
**Pull request by vcolin7**

### CODE REVIEW

**Comment by vcolin7** [comment]
Overrode Check Enforcer as it incorrectly flagged [an issue](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1193141&view=logs&j=36df4387-3551-57fa-098c-b43a23930261&t=3d911a90-41b0-55ea-c0ca-a4562a31c54b&l=25) related to the `eng/common` folder.

File: sdk/keyvault/azure-security-keyvault-secrets/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Maven dependency for the Azure Key Vault Secrets client library. Add it to your
 <dependency>
     <groupId>com.azure</groupId>
     <artifactId>azure-security-keyvault-secrets</artifactId>
-    <version>4.3.4</version>
+    <version>4.4.0-beta.3</version>
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-secrets/README.md:19]: README versions should point to stable.

File: sdk/keyvault/azure-security-keyvault-administration/README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Maven dependency for the Azure Key Vault Administration library. Add it to your
 <dependency>
     <groupId>com.azure</groupId>
     <artifactId>azure-security-keyvault-administration</artifactId>
-    <version>4.0.4</version>
+    <version>4.1.0-beta.3</version>
```
Comment (g2vinay) [inline sdk/keyvault/azure-security-keyvault-administration/README.md:17]: README version should point to stable.

File: sdk/keyvault/azure-security-keyvault-secrets/README.md (hunk 1 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-secrets/README.md:19]: I think the ones that will be included with a beta release's tag should point to the beta version, after releasing said version I'll update the READMEs to point to stable.

File: sdk/keyvault/azure-security-keyvault-administration/README.md (hunk 2 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/README.md:17]: I think the ones that will be included with a beta release's tag should point to the beta version, after releasing said version I'll update the READMEs to point to stable.

**Review by g2vinay — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-administration/README.md (hunk 2 above)
Comment (g2vinay) [reply sdk/keyvault/azure-security-keyvault-administration/README.md:17]: sounds good

**Pull request merged**